File: dev-docs/snippets/documents/wait-for-write-confirmation.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Wait until the server confirms it has all of this client's writes.
export async function waitForWriteConfirmation(
  client: JsBaoClient,
  documentId: string,
) {
  // #region example
  const confirmed = await client.documents.waitForWriteConfirmation(documentId);
  // #endregion example
  return confirmed;
}

File: dev-docs/snippets/documents/wait-for-write-confirmation.swift
import JsBaoClient

// Wait until the server confirms it has all of this client's writes. Returns a
// Bool — true once confirmed, false on timeout (it does not throw).
func waitForWriteConfirmation(client: JsBaoClient, documentId: String) async {
  // #region example
  let confirmed = await client.documents.waitForWriteConfirmation(documentId: documentId)
  // #endregion example
  _ = confirmed
}
